packages/ember-application/lib/system/engine-instance.js

```diff
@@ -154,6 +159,32 @@ if (isEnabled('ember-application-engines')) {
       setEngineParent(engineInstance, this);
 
       return engineInstance;
+    },
+
+    /**
+      Clone dependencies shared between an engine instance and its parent.
+
+      @private
+      @method cloneParentDependencies
+    */
+    cloneParentDependencies() {
+      let parent = getEngineParent(this);
+
+      [
+        'route:basic',
+        'event_dispatcher:main',
+        P`-bucket-cache:main`,
+        'service:-routing'
+      ].forEach((key) => {
+        this.register(key, parent.resolveRegistration(key));
```

Comment:
@rwjblue I wasn't properly mocking the engine's parent in some instance initializer tests. Should be fixed now.